src/adapt.ts

```diff
@@ -41,6 +41,22 @@ export function adaptInput(functionName: string, input: any, inputSpecs: starkne
     const adapted: string[] = [];
     let lastSpec: starknet.Argument = { type: null, name: null };
 
+    // User won't pass array length as an argument, so subtract the number of array elements to the expected amount of arguments
+    const countArrays = inputSpecs.filter( i => i.name.endsWith(LEN_SUFFIX)).length;
```

Comment:
This will not always be correct, e.g. in case an input argument has the len suffix, but does not precede an actual array. A more complete check can be found in this same function (line 68). Also I think the current implementation allows passing the len argument, although not as a required argument.